ddtrace/contrib/openai/patch.py

```diff
@@ -244,6 +247,72 @@ def record_usage(self, span, usage):
             span.set_metric("openai.response.usage.%s_tokens" % token_type, num_tokens)
             self._statsd.distribution("tokens.%s" % token_type, num_tokens, tags=tags)
 
+    def generate_completion_llm_records(self, resp, span, args, kwargs):
+        # type: (Any, Span, List[Any], Dict[str, Any]) -> None
+        """Generate payloads for the LLM Obs API from a completion."""
+        if not self._config.llmobs_enabled:
+            return
+        choices = resp.choices
+        n = kwargs.get("n", 1)
+        prompt = kwargs.get("prompt", "")
+        if isinstance(prompt, str):
+            prompt = [prompt]
+        # Note: LLMObs ingest endpoint only accepts a 1:1 prompt-response mapping per record,
+        #  so we need to deduplicate and send unique prompt-response records if n > 1.
+        for i in range(n):
+            unique_choices = choices[i::n]
```

Comment:
So we're getting a slice that starts at index i and skips n values? So if n=3 choices=[A, B, C, D, E],
 iteration 1 -> unique_choices=[A, D], iteration 2 -> [B, E], iteration 3 -> [C]. Is this expected? This algorithm just seems funky.

Reply:
The way openai completions work is that you send P number of prompts, specify N number of choices to return per prompt, meaning you will receive P * N choices in the response.

Let's say you send openai 3 prompts: [A, B, C], and specify N=2. Then you'll receive a completions in the form
```
[A1, A2, B1, B2, C1, C2]
```
Where A1 corresponds to the first response created for prompt A, A2 corresponds to the second response created for prompt A, and etc.

This algorithm is just making sure that we create a record with deduplicated, unique prompt-response pairs in the following form:
```
[A1, B1, C1], [A2, B2, C2]
```
i.e. this loop is just starting at index `i` and iterating over every `n`th choice which will correspond to a different prompt.